main.js

```diff
@@ -0,0 +1,165 @@
+  conditions.default = searchLocation.zip === getDefaultLocation().zip //
+  return conditions;
+}
+
+var renderWeatherView = function() {
+  $('#weather-view').empty();
+  var weatherViewHTML = weatherViewTemplate(weatherConditions);
+  var mapHTML = mapTemplate(setGoogleString());
+  $('#weather-view').append(weatherViewHTML, mapHTML);
+
+};
+var renderForecastView = function() {
+  $('#forecast-view').empty();
+  var forecastViewHTML = forecastViewTemplate(forecastData);
+  $('#forecast-view').append(forecastViewHTML);
+}
+
+
+var fetchWeather = function() {
+  var queryString = setQueryString();
+  var url = `http://api.openweathermap.org/data/2.5/weather?${queryString}&units=imperial&appid=6c2fa6a0fe784b355d9a286151f346b6`
+  $.ajax({
+    method: "GET",
+    url: url,
+    dataType: "json",
+    success: function(data) {
+      weatherConditions = setConditions(data);
+      renderWeatherView();
+    },
+
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log(errorThrown, textStatus)
+    }
+  })
+}
+
+var fetchForecast = function() {
+  var queryString = setQueryString();
+  var url = `http://api.openweathermap.org/data/2.5/forecast?${queryString}&units=imperial&appid=6c2fa6a0fe784b355d9a286151f346b6`
+  $.ajax({
+    method: "GET",
+    url: url,
+    dataType: "json",
+    success: function(data) {
+      //empty forecasts array before adding new data
+      while (forecastData.forecasts.length) {
+        forecastData.forecasts.pop();
+      };
+      var fiveDayArray = data.list.filter(function(forecast, index) {
+        return forecast.dt_txt.indexOf('21:00') !== -1; //gives you forecasts for 9pm UTC/4pm ET/1pm PT
+      })
+      fiveDayArray.forEach(function(day) {
+        var forecastForDay = setConditions(day);
+        forecastForDay.day = moment(day.dt_txt).format('dddd');
+        forecastData.forecasts.push(forecastForDay);
+      })
+      renderForecastView();
+    },
+
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log(errorThrown, textStatus)
+    }
+  })
+}
+
+getLocation();
+
+if (searchLocation.zip || searchLocation.latitude) {
+  fetchWeather();
+  fetchForecast();
+}
+
+//when user enters search, check that city is a valid us zipcode. if valid, set searchZip equal to .val
+$('#search').on('click', function() {
+  var userInput = $('#search-zip').val();
+  if (userInput.length === 5 && userInput == parseInt(userInput)) {
+    searchLocation.zip = userInput;
+    fetchWeather();
+    fetchForecast();
+  } else {
+    alert("Please enter a 5-digit US zipcode.")
+  }
+})
+$('#weather-view').on('click', '#set-default', function() {
+  setDefaultLocation();
+  $(this).html('&#x2714; City set as default')
+})
+
+$('#use-location').on('click', function() {
+  if (searchLocation.latitude && searchLocation.longitude) {
+    searchLocation.zip = null;
+    fetchWeather();
+    fetchForecast();
+  } else {
+    console.log('unable to get location')
```

Comment:
Good job handling the error, but why not use alert() or put it on the page... it's a pretty bad experience as is.